service/authorization/authorization.go

```diff
@@ -57,28 +65,59 @@ func NewRegistration() serviceregistry.Registration {
 		Namespace:   "authorization",
 		ServiceDesc: &authorization.AuthorizationService_ServiceDesc,
 		RegisterFunc: func(srp serviceregistry.RegistrationParams) (any, serviceregistry.HandlerServer) {
+			var (
+				err             error
+				entitlementRego []byte
+				authZCfg        = new(Config)
+			)
 			// default ERS endpoint
-			as := &AuthorizationService{eng: srp.Engine, sdk: srp.SDK, logger: srp.Logger}
+			as := &AuthorizationService{sdk: srp.SDK, logger: srp.Logger}
 			if err := srp.RegisterReadinessCheck("authorization", as.IsReady); err != nil {
 				slog.Error("failed to register authorization readiness check", slog.String("error", err.Error()))
 			}
 
-			authZCfg := Config{
-				ERSURL:        "http://localhost:8080/entityresolution/resolve",
-				ClientID:      "tdf-authorization-svc",
-				ClientSecret:  "secret",
-				TokenEndpoint: "http://localhost:8888/auth/realms/opentdf/protocol/openid-connect/token",
+			// Load Defaults
+			if err := defaults.Set(authZCfg); err != nil {
+				panic(fmt.Errorf("failed to set defaults for authorization service config: %w", err))
 			}
+
 			if err := mapstructure.Decode(srp.Config.ExtraProps, &authZCfg); err != nil {
 				panic(fmt.Errorf("invalid auth svc cfg [%v] %w", srp.Config.ExtraProps, err))
 			}
 
-			config := clientcredentials.Config{ClientID: authZCfg.ClientID, ClientSecret: authZCfg.ClientSecret, TokenURL: authZCfg.TokenEndpoint}
-			slog.Debug("authorization service client config", slog.Any("config", config))
-			newTokenSource := oauth2.ReuseTokenSourceWithExpiry(nil, config.TokenSource(context.Background()), tokenExpiryDelay)
-			slog.Debug("authorization service token source created", slog.Any("token_source", newTokenSource))
+			slog.Debug("authorization service config", slog.Any("config", authZCfg))
```

Comment:
Did we end up deploying a global logging solution for masking config secrets in logs?

Reply:
No it was only implemented for the root config object and I don't think its working last time I looked.

Reply:
https://github.com/opentdf/platform/issues/751